### PR TITLE
fix(apps/pano): "New post" button appears without login

### LINKED ISSUE #677: bug(apps/kampus): new post button appears without login
### Describe the bug

We need to check session status in `/post/create` route and for `New Post` button on home page.

### Steps to reproduce the behavior

Visit https://pano.dev.kamp.us or https://pano.dev.kamp.us/post/create 

### Expected behavior

_No response_

### PR DESCRIPTION
**Pull request by emredevsalot**
# Description

I'm 95% sure this is not the desired way of implementing this. However I wanted to get it working first, and continue from there.

I couldn't get the `session` directly at `apps/kampus/app/pano/PanoFeed.tsx`, and I got the following error.

`@kampus-apps/kampus:dev: [next-auth][warn][EXPERIMENTAL_API] 
@kampus-apps/kampus:dev: getServerSession is used in a React Server Component. 
@kampus-apps/kampus:dev: https://next-auth.js.org/configuration/nextjs#getServerSession}
@kampus-apps/kampus:dev: https://next-auth.js.org/warnings#EXPERIMENTAL_API`

I looked into it, one suggestion was to call `session` with `authOptions`(await getServerSession(authOptions)), but it didn't accept the `authOptions`.

What do you suggest me to delve into research-wise to solve the problem? 

### Checklist

- [x] discord username: `emredevsalot`
- [x] Closes #677
- [ ] PR must be created for an issue from issues under "In progress" column from [our project board](https://github.com/orgs/kamp-us/projects/2/views/1).
- [ ] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules, and should include the app/package/service name in the title. For example, a title like "docs(@kampus-apps/pano): Add README.md" can be used.
- [ ] Related file selection: Only relevant files should be touched and no other files should be affected.
- [ ] I ran `npx turbo run` at the root of the repository, and build was successful.
- [ ] I installed the npm packages using `npm install --save-exact <package>` so my package is pinned to a specific npm version. Leave empty if no package was installed. Leave empty if no package was installed with this PR.

### How were these changes tested?

Please describe the tests you did to test the changes you made. Please also specify your test configuration.

### CODE REVIEW

**Comment by FurkanGM** [comment]
can u fix eslint errors?

**Comment by cansirin** [comment]
Thank you so much for the start and development!

instead of using the Session, we have to check if Actor query is returning null or the actor itself.

**Comment by FurkanGM** [comment]
> @kampus-apps/kampus:dev: [next-auth][warn][EXPERIMENTAL_API]  @kampus-apps/kampus:dev: getServerSession is used in a React Server Component.  @kampus-apps/kampus:dev: https://next-auth.js.org/configuration/nextjs#getServerSession} @kampus-apps/kampus:dev: https://next-auth.js.org/warnings#EXPERIMENTAL_API

The console output is not an error; there is no problem. It's simply a warning because you used experimental function `getServerSession`.

File: apps/kampus/app/pano/__generated__/PanoFeed_viewer.graphql.ts [hunk 1]
```diff
@@ -11,6 +11,9 @@
 import { Fragment, ReaderFragment } from 'relay-runtime';
 import { FragmentRefs } from "relay-runtime";
 export type PanoFeed_viewer$data = {
+  readonly actor: {
+    readonly displayName: string | null;
+  } | null;
```
Comment (emredevsalot) [inline apps/kampus/app/pano/__generated__/PanoFeed_viewer.graphql.ts:16]: I just saw this is in __generated__ folder, I didn't see that while working on it and thought I was going to write this here myself. After that, I've used npm run codegen, I don't know the step-by-step process.

How am I going to get the actor in the `apps/kampus/app/pano/PanoFeed.tsx` file after adding  
`actor {displayName}` to the fragment?

Comment (Ketcap) [reply apps/kampus/app/pano/__generated__/PanoFeed_viewer.graphql.ts:16]: you need to run relay see this line https://github.com/kamp-us/monorepo/blob/aa6e9c6a1871c4f7d341082c4c9679f1acbb6606/apps/kampus/package.json#L10

File: apps/kampus/app/pano/PanoFeed.tsx [hunk 2]
```diff
@@ -35,6 +35,9 @@ const fragment = graphql`
 const viewerFragment = graphql`
   fragment PanoFeed_viewer on Viewer {
     ...PostItem_viewer
+    actor {
+      displayName
```
Comment (usirin) [inline apps/kampus/app/pano/PanoFeed.tsx:39]:
```suggestion
      ... on User {
        dislayName
      }
```

File: apps/kampus/app/pano/__generated__/PanoFeed_viewer.graphql.ts (hunk 1 above)
Comment (usirin) [reply apps/kampus/app/pano/__generated__/PanoFeed_viewer.graphql.ts:16]: This is fine, no worries.

File: apps/kampus/app/pano/PanoFeed.tsx [hunk 3]
```diff
@@ -49,6 +52,8 @@ export function PanoFeed(props: Props) {
     props.panoFeed
   );
   const viewer = useFragment(viewerFragment, props.panoViewer);
+  // I get undefined here.
+  console.log(viewer?.actor);
```
Comment (usirin) [inline apps/kampus/app/pano/PanoFeed.tsx]: if you're not logged in, it's normal for it to be undefined.

**Comment by cansirin** [comment]
Ideally with my #665 PR, I will provide a hook for fetching the viewer and we can start using this in the components.

**Comment by emredevsalot** [comment]
When I tried to check if I was getting the actor state correctly, I tested it with console.log. Whenever I refresh the page, it logs the actor 6 times, is that normal? Is it something we come across only in local? (I run the console.log in the PanoFeed function, right before the return statement.)
image

**Comment by usirin** [comment]
@emredevsalot now that we have #694 merged, let's modify this PR to use `useViewer` hook created.

**Comment by cansirin** [comment]
using hook was kind of against to relay mentality, for that reason we are not going to use useViewer hook anymore, we gotta keep continue spreading fragments and transport viewer ref for now.
cc @usirin

**Pull request closed without merging**